nan_json.h

```diff
@@ -0,0 +1,160 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2017 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#ifndef NAN_JSON_H_
+#define NAN_JSON_H_
+
+#if NODE_MODULE_VERSION < NODE_0_12_MODULE_VERSION
+#define NAN_JSON_H_NEED_PARSE 1
+#else
+#define NAN_JSON_H_NEED_PARSE 0
+#endif  // NODE_MODULE_VERSION < NODE_0_12_MODULE_VERSION
+
+#if NODE_MODULE_VERSION > NODE_6_0_MODULE_VERSION
+#define NAN_JSON_H_NEED_STRINGIFY 0
+#else
+#define NAN_JSON_H_NEED_STRINGIFY 1
+#endif  // NODE_MODULE_VERSION > NODE_6_0_MODULE_VERSION
+
+class JSON {
+ public:
+  JSON() {
+#if NAN_JSON_H_NEED_PARSE + NAN_JSON_H_NEED_STRINGIFY
+    Nan::MaybeLocal<v8::Value> maybe_global_json = Nan::Get(
+      Nan::GetCurrentContext()->Global(),
+      Nan::New("JSON").ToLocalChecked()
+    );
+
+    if (!maybe_global_json.IsEmpty()) {
```

Comment:
At first, I didn't think it would ever be possible for this to be empty. I included the check because it was a `MaybeLocal` and thus needed to be checked, although now that you mention it, yes, a user can indeed overwrite the `JSON` object from `JavaScript`.  Part of me thinks that if the user has done this, that they deserve whatever they get as a result 3:-)  But perhaps it would be a good idea to throw an error or add an assertion.  I'm just not sure.  Without changing the code, if this does end up being empty somehow, the entry point method (`Parse` / `Stringify`) will simply return `Undefined()`.  To me, this seems the most compatible path.